src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/attn_quant.cpp

```diff
@@ -170,6 +176,223 @@ static void quant_u8(const T* src, uint8_t* dst, size_t n, float& scale, float&
+        first_half = _mm512_slli_epi32(first_half, 4);
+        auto mask = _mm512_set1_epi32(0x0F);
+        second_half = _mm512_and_epi32(second_half, mask);
+        auto combined = _mm512_or_epi32(first_half, second_half);
+        _mm512_mask_cvtepi32_storeu_epi8(dst_ptr + i / 2, 0xffff, combined);
+    }
+#endif
+#if defined(HAVE_AVX2)
+    auto v256_zero = _mm256_set1_epi32(0);
+    auto v256_upper = _mm256_set1_epi32(15);
+    auto v256_scale = _mm256_set1_ps(1 / scale);
+    auto v256_zp = _mm256_set1_ps(zp);
+    for (; i + vec_len_f32_avx2 * 2 <= n; i += vec_len_f32_avx2 * 2) {
+        auto v0 = mm256_uni_loadu_ps(src + i);
+        auto v1 = mm256_uni_loadu_ps(src + i + vec_len_f32_avx2);
+        v0 = _mm256_fmadd_ps(v0, v256_scale, v256_zp);
+        v1 = _mm256_fmadd_ps(v1, v256_scale, v256_zp);
+        v0 = _mm256_round_ps(v0, _MM_ROUND_NEAREST);
+        v1 = _mm256_round_ps(v1, _MM_ROUND_NEAREST);
+
+        auto v0_i32 = _mm256_cvtps_epi32(v0);
+        auto v1_i32 = _mm256_cvtps_epi32(v1);
+        v0_i32 = _mm256_max_epi32(v0_i32, v256_zero);
+        v1_i32 = _mm256_max_epi32(v1_i32, v256_zero);
+        v0_i32 = _mm256_min_epi32(v0_i32, v256_upper);
+        v1_i32 = _mm256_min_epi32(v1_i32, v256_upper);
+        auto idx1 = _mm256_set_epi32(7, 5, 3, 1, 6, 4, 2, 0);
+        v0_i32 = _mm256_permutevar8x32_epi32(v0_i32, idx1);
+        v1_i32 = _mm256_permutevar8x32_epi32(v1_i32, idx1);
+        //    0,1,2,3,4,5,6,7 | 8,9,10,11,12,13,14,15
+        //       _mm256_permutevar8x32_epi32
+        //    0,2,4,6,1,3,5,7 | 8,10,12,14,9,11,13,15
+        //       _mm256_permute2x128_si256
+        // 0,2,4,6,8,10,12,14 | 1,3,5,7,9,11,13,15
+        //          shift + mask + or
+        //     [0,1],[2,3], ..., [12,13], [14,15]
+        auto first_half = _mm256_permute2x128_si256(v0_i32, v1_i32, 0x20);
+        auto second_half = _mm256_permute2x128_si256(v0_i32, v1_i32, 0x31);
+        first_half = _mm256_slli_epi32(first_half, 4);
+        auto mask = _mm256_set1_epi32(0x0F);
+        second_half = _mm256_and_si256(second_half, mask);
+        auto combined = _mm256_or_si256(first_half, second_half);
+
+        auto high4 = _mm256_extractf128_si256(combined, 1);
+        auto low4 = _mm256_castsi256_si128(combined);
+        // ignore sign bit for u4 case
+        auto packed = _mm_packus_epi32(low4, high4);
+        packed = _mm_packus_epi16(packed, packed);
+        _mm_storel_epi64(reinterpret_cast<__m128i*>(dst_ptr + i / 2), packed);
+    }
+#endif
+    for (; i < n; i++) {
+        float tmp = src[i];
+#define MIN(a, b) ((a) < (b) ? (a) : (b))
+        uint8_t src_val = MIN(15, (uint8_t)(std::round(tmp / scale + zp)));
+        uint8_t dst_val = i % 2 == 0 ? 0 : dst_ptr[i / 2];
+        dst_val = insert_half_byte(dst_val, src_val, (uint8_t)(i % 2));
+        dst_ptr[i / 2] = dst_val;
+    }
+}
+
+template <typename T>
+static void quant_i4(const T* src, void* dst, size_t n, float& scale) {
```

Comment:
So it makes sense to remove i4 instances now, isn't it?

Reply:
Sure, let me remove s4 cases since it's not used now.